### PR TITLE
Update third_party/musl subtree.

### PR DESCRIPTION
**Pull request by briantting**

### CODE REVIEW

**Comment by briantting** [comment]
```
../../starboard/string.h:234:43: error: use of undeclared identifier '__LONG_MAX'
  int result = SbStringFormat(out_buffer, SSIZE_MAX, format, arguments);
                                          ^
../../third_party/musl/include/limits.h:53:19: note: expanded from macro 'SSIZE_MAX'
#define SSIZE_MAX LONG_MAX
                  ^
../../third_party/musl/include/limits.h:29:18: note: expanded from macro 'LONG_MAX'
#define LONG_MAX __LONG_MAX
```

seems like theres a missing `__LONG_MAX` that needs to be defined in `bits/alltypes.h`?

**Comment by briantting** [comment]
ran `generate.sh` script in third_party/musl to regenerate the `alltypes.h` files

File: third_party/musl/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,519 @@
+# Copyright 2021 The Cobalt Authors. All Rights Reserved.
```
Comment (kaidokert) [inline third_party/musl/BUILD.gn]: nitpick: 2023 ?

File: third_party/musl/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+
+if (target_cpu == "x64") {
+  musl_arch = "x86_64"
+} else if (target_cpu == "x86") {
+  musl_arch = "i386"
+} else if (target_cpu == "arm") {
+  musl_arch = "arm"
+} else if (target_cpu == "arm64") {
+  musl_arch = "aarch64"
+} else if (target_cpu == "mips") {
```
Comment (kaidokert) [inline third_party/musl/BUILD.gn]: nitpick: we aren't supporting mips so technically this isn't needed.

File: third_party/musl/BUILD.gn [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+}
+
+target(gtest_target_type, "musl_unittests") {
+  testonly = true
+  sources = [ "test/type_size_test.cc" ]
+  deps = [
+    ":c",
+    "//starboard:starboard_headers_only",
+  ]
+}
```
Comment (kaidokert) [inline third_party/musl/BUILD.gn:540]: missing endline .. not sure why pre-commit isn't catching it ? you can run it on a single commit if you need with `pre-commit run --from-ref=<hash> --to-ref=HEAD`

File: third_party/musl/METADATA [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    value: "git://git.musl-libc.org/musl"
+  }
+  version: "f5f55d6589940fd2c2188d76686efe3a530e64e0"
+  last_upgrade_date {
+    year: 2023
+    month: 5
+    day: 1
+  }
+  license_type: NOTICE
+}
```
Comment (kaidokert) [inline third_party/musl/METADATA:17]: missing endline - again pre-commit should be catching it

File: third_party/musl/src/starboard/errno/__errno_location.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
+#include <errno.h>
```
Comment (kaidokert) [inline third_party/musl/src/starboard/errno/__errno_location.c]: I think we are missing Cobalt copyright header here. Probably everything under musl/src/starboard should have it.

But if those weren't here before it's fine to submit as is

**Review by kaidokert — APPROVED** [review]
It's mostly fine as is i think - please fix the missing endlines but the old code that you din't write can be left as is.

Also i realize pre-commit doesn't run on these files by default.

**Pull request merged**